tests/integration/models/test_poromechanics.py

```diff
@@ -50,51 +50,52 @@ def initial_condition(self):
 
         """
         super().initial_condition()
-        # Initial pressure equals reference pressure (defaults to zero).
-        self.equation_system.set_variable_values(
-            self.fluid.pressure() * np.ones(self.mdg.num_subdomain_cells()),
-            [self.pressure_variable],
-            to_state=True,
-            to_iterate=True,
-        )
-        sd, sd_data = self.mdg.subdomains(return_data=True)[0]
-        # Initial displacement.
-        if len(self.mdg.subdomains()) > 1:
-
-            top_cells = sd.cell_centers[1] > 0.5
-            vals = np.zeros((self.nd, sd.num_cells))
-            vals[1, top_cells] = self.fluid.convert_units(0.042, "m")
+        if self.time_manager.is_init():
```

Comment:
I will have to put my answer ti this on hold. I had guessed that the answer to your question is "yes" and one could omit the conditional which I believe I added early in the test-driven development. However, when removing it (and this is directly connected or maybe even more reasoned by the use of super().initial_condition() which also invokes the restart. Since I have not the required overview of the implications what which state changes in the discretization etc. I will have to go in more details and debug again. I have a slightly bad feeling that there is a high risk in the current approach that one is not doing what intended if not testing the restart for each particular model, because it seems the effects can be quite hidden. I will report back.

Reply:
After separating restart from intial_condition, I agree that this was too hardcoded and should not be the foreseen practice.